packages/vizbuilder/src/helpers/chartconfig.js

```diff
@@ -0,0 +1,330 @@
+import {
+  BarChart,
+  Donut,
+  Geomap,
+  LinePlot,
+  StackedArea,
+  Treemap
+} from "d3plus-react";
+import {formatAbbreviate} from "d3plus-format";
+import {sortByGeographyState, sortByCustomKey} from "./sorting";
+import { getAreaChartDimensions } from "./constants";
+
+export const charts = {
+  barchart: BarChart,
+  barchartyear: BarChart,
+  // histogram: BarChart,
+  donut: Donut,
+  geomap: Geomap,
+  lineplot: LinePlot,
+  stacked: StackedArea,
+  treemap: Treemap
+};
+
+export const ALL_YEARS = "All years";
+
+export const legendConfig = () => ({
+  label: false,
+  shapeConfig: {
+    width: 0,
+    height: 0
+  }
+});
+
+export const tooltipConfig = (query, keyTitle, keyBody) => {
+  keyTitle = keyTitle || query.drilldown.name;
+  keyBody = keyBody || query.measure.name;
+
+  const config = {
+    width: 60,
+    title: d =>
+      `<h5 class="title xs-small">${[].concat(d[keyTitle]).join("<br/>")}</h5>`
+  };
+
+  if (query.timeDrilldown && keyTitle !== "Year") {
```

Comment:
rewrite this big `if` statement to use the tooltip's built-in table, and conditionally add more arrays to the list if the values (like moe and year) are present. here is an example usage: https://d3plus.org/examples/d3plus-tooltip/getting-started/